### PR TITLE
Use StateBuffer instead of IntraBlockState for consensus tests

### PR DESCRIPTION
**Pull request by yperbasis**
(since IntraBlockState is supposed to be used inside a single block, not across blocks).

Also fix a corner case by introducing `StorageCreateDelta`.

### CODE REVIEW

File: cmd/consensus.cpp [hunk 1]
```diff
@@ -215,32 +215,42 @@ static void check_rlp_err(rlp::DecodingError err) {
...
-void init_pre_state(const nlohmann::json& pre, IntraBlockState& state) {
+void init_pre_state(const nlohmann::json& pre, StateBuffer& state) {
     for (const auto& entry : pre.items()) {
         evmc::address address{to_address(*from_hex(entry.key()))};
-        const nlohmann::json& account{entry.value()};
-        Bytes balance_str{*from_hex(account["balance"].get<std::string>())};
+        const nlohmann::json& j{entry.value()};
+
+        Account account;
+        Bytes balance_str{*from_hex(j["balance"].get<std::string>())};
```
Comment (AndreaLanfranchi) [inline cmd/consensus.cpp:224]: Question ... from_hex returns std::optional<Bytes> ... are we sure of the sanity of the Json payload ?

File: cmd/consensus.cpp [hunk 2]
```diff
@@ -284,56 +293,58 @@ Status run_block(const nlohmann::json& b, const ChainConfig& config, IntraBlockS
...
-        Bytes balance_str{*from_hex(account["balance"].get<std::string>())};
+        Bytes balance_str{*from_hex(j["balance"].get<std::string>())};
         auto [expected_balance, err1]{rlp::read_uint256(balance_str, /*allow_leading_zeros=*/true)};
         check_rlp_err(err1);
-        intx::uint256 actual_balance{state.get_balance(address)};
-        if (actual_balance != expected_balance) {
+        if (account->balance != expected_balance) {
             std::cout << "Balance mismatch for " << entry.key() << ":\n";
-            std::cout << to_string(actual_balance, 16) << " ≠ " << account["balance"] << "\n";
+            std::cout << to_string(account->balance, 16) << " ≠ " << j["balance"] << "\n";
```
Comment (AndreaLanfranchi) [inline cmd/consensus.cpp:323]: Please use ASCII chars as we get different display output on windows.
I'd go for `!=` instead of `≠`

**Review by AndreaLanfranchi — CHANGES_REQUESTED** [review]

File: cmd/consensus.cpp (hunk 2 above)
Comment (yperbasis) [reply cmd/consensus.cpp]: Done.

File: cmd/consensus.cpp (hunk 1 above)
Comment (yperbasis) [reply cmd/consensus.cpp]: Right, I've replaced it with checked access to be on the safe side.

**Review by AndreaLanfranchi — APPROVED** [review]
